### PR TITLE
Add transfer buttons

### PR DESCRIPTION
**Pull request by benn02**
### PR Checklist

Please make sure to fulfil the following conditions before marking this PR ready for review:

-   [x] If this PR adds or changes features or fixes bugs, this has been added to the changelog
-   [x] If this PR adds new actions or other ways to alter the state, [test scenarios](https://github.com/hpi-sam/digital-fuesim-manv-public-test-scenarios) have been added

### CODE REVIEW

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/start-transfer.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    TransferPoint,
+    UUIDSet,
+    Vehicle,
+} from 'digital-fuesim-manv-shared';
+import { Subject } from 'rxjs';
+
+export interface TransferOptions {
+    vehicleToTransfer: Vehicle | undefined;
+    patientsToTransfer: UUIDSet | undefined;
+    transferDestination: Hospital | TransferPoint | undefined;
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/start-transfer.service.ts:13]:
```suggestion
    vehicleToTransfer?: Vehicle;
    patientsToTransfer?: UUIDSet;
    transferDestination?: Hospital | TransferPoint;
```

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts [hunk 2]
```diff
@@ -34,4 +37,14 @@ export class SimulatedRegionOverviewPatientInteractionBarComponent {
         });
         this.selectPatientService.selectPatient('');
     }
+
+    public initiatePatientTransfer() {
+        const patientsToTransfer: Mutable<UUIDSet> = {};
+        patientsToTransfer[this.patientId] = true;
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts:43]:
```suggestion
        const patientsToTransfer: Mutable<UUIDSet> = {
            [this.patientId]: true,
        };
```

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts [hunk 3]
```diff
@@ -41,12 +44,28 @@ export class SimulatedRegionOverviewBehaviorTabComponent
         private readonly store: Store<AppState>
     ) {}
 
-    ngOnInit(): void {
+    async ngOnInit(): Promise<void> {
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts:47]:
```suggestion
    async ngOnInit() {
```

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.ts [hunk 4]
```diff
@@ -119,6 +132,17 @@ export class SimulatedRegionOverviewBehaviorTransferVehiclesComponent
     }
 
     ngOnInit(): void {
+        if (this.transferToStart) {
+            this.transferCollapsed = false;
+            this.vehicleToSend = this.transferToStart.vehicleToTransfer;
+            this.selectedDestination = this.transferToStart.transferDestination;
+            this.clickedPatients =
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.ts:139]: These three variable names should be similar, to make it clear, that they belong together.
How about `selectedVehicle`, `selectedDestination` and `selectedPatients`?

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.ts [hunk 5]
```diff
@@ -391,4 +373,63 @@ export class SimulatedRegionOverviewBehaviorTransferVehiclesComponent
...
+                            .includes(patientId)
+                    ) {
+                        if (clicked) {
+                            this.minPatients--;
+                        }
+                        delete this.clickedPatients[patientId];
+                    }
+                }
+            );
+        });
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.ts:433]: Why is this needed?

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts (hunk 2 above)
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts:43]:
```suggestion
        const patientsToTransfer: Mutable<UUIDSet> = {
            [this.patientId]: true,
        };
```

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/overview/simulated-region-overview.component.html [hunk 6]
```diff
@@ -90,6 +90,7 @@
                     ><app-simulated-region-overview-behavior-tab
                         class="mh-100 d-block h-100"
                         [simulatedRegion]="(simulatedRegion$ | async)!"
+                        [transferToStart]="transferToStart"
```
[the remark this answers is not in the document]
Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/overview/simulated-region-overview.component.html:93]: Done

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/start-transfer.service.ts (hunk 1 above)
Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/start-transfer.service.ts:13]: Done

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts (hunk 3 above)
Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts:47]: Done

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.ts (hunk 4 above)
Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.ts:139]: Done

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts (hunk 2 above)
Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts:43]: Done

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts [hunk 7]
```diff
@@ -41,12 +44,28 @@ export class SimulatedRegionOverviewBehaviorTabComponent
...
+            this.selectedBehavior = this.simulatedRegion.behaviors.find(
+                (behavior) => behavior.type === 'transferBehavior'
+            );
+            if (!this.selectedBehavior) {
+                await this.exerciseService.proposeAction({
+                    type: '[SimulatedRegion] Add Behavior',
+                    simulatedRegionId: this.simulatedRegion.id,
+                    behaviorState:
+                        simulationBehaviorDictionary.transferBehavior.behaviorState.create(),
+                });
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts:63]: You should add something like:
```suggestion
                const response = await this.exerciseService.proposeAction({
                    type: '[SimulatedRegion] Add Behavior',
                    simulatedRegionId: this.simulatedRegion.id,
                    behaviorState:
                        simulationBehaviorDictionary.transferBehavior.behaviorState.create(),
                });
                if (!response.success) return
```

**Review by Greenscreen23 — CHANGES_REQUESTED** [review]

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts (hunk 7 above)
Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts:63]: I do not see why I would return if I couldn't add the behavior... The current Code the switches to no behavior at all what is expected. A return would just crash the application in that case. One could then try to go to the globalLastBehavior . But I wouldn't suggest that since that would seem even more unintuitive than showing no behavior

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/vehicles-tab/simulated-region-overview-vehicles-tab.component.ts [hunk 8]
```diff
@@ -175,4 +177,12 @@ export class SimulatedRegionOverviewVehiclesTabComponent implements OnInit {
         );
         return index === -1 ? vehicleTemplates.length : index;
     }
+
+    public initiateVehicleTransfer(vehicle: Vehicle) {
+        this.startTransferService.initiateNewTransferFor({
+            vehicleToTransfer: vehicle,
+            patientsToTransfer: vehicle.patientIds,
+            transferDestination: undefined,
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/vehicles-tab/simulated-region-overview-vehicles-tab.component.ts:185]:
```suggestion
            vehicleToTransfer: vehicle,
            patientsToTransfer: vehicle.patientIds,
```

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts [hunk 9]
```diff
@@ -34,4 +37,15 @@ export class SimulatedRegionOverviewPatientInteractionBarComponent {
...
     }
+
+    public initiatePatientTransfer() {
+        const patientsToTransfer: Mutable<UUIDSet> = {
+            [this.patientId]: true,
+        };
+        this.startTransferService.initiateNewTransferFor({
+            vehicleToTransfer: undefined,
+            patientsToTransfer,
+            transferDestination: undefined,
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/patients-tab/simulated-region-overview-patient-interaction-bar/simulated-region-overview-patient-interaction-bar.component.ts:48]:
```suggestion
            patientsToTransfer,
```

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts (hunk 7 above)
Comment (Greenscreen23) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/simulated-region-overview-behavior-tab.component.ts:63]: Ah, makes sense

File: frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.html [hunk 10]
```diff
@@ -304,118 +304,130 @@ <h4 class="ps-0 pe-0 text-nowrap">
...
-                            (reachableTransferPoints.length +
-                                reachableHospitals.length ===
-                                1 &&
-                                selectedDestination)
-                        "
+                    <div
+                        ngbDropdown
+                        placement="bottom-start"
+                        autoClose="outside"
+                        class="d-inline-block overflow-visible text-center c"
```
Comment (Greenscreen23) [inline frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.html:321]: What is the class `c`?

Comment (benn02) [reply frontend/src/app/pages/exercises/exercise/shared/simulated-region-overview/tabs/behavior-tab/behaviors/transfer-vehicles/simulated-region-overview-behavior-transfer-vehicles.component.html:321]: Idk bro

**Review by Greenscreen23 — APPROVED** [review]

**Review by Greenscreen23 — APPROVED** [review]
